File: src/baekjoon/step/stackQueueDeque/PROB10773.py
# 재현이는 잘못된 수를 부를 때마다 0 을 외쳐서, 가장 최근에 재민이가 쓴 수를 지우게 한다.
# 재민이는 이렇게 모든 수를 받아 적은 후 그 수의 합을 알고 싶어 한다.
# 재민이를 도와주자!


import sys

# 수의 개수 입력받기
K = int(input())

# 스택 리스트 생성
stack =[]

# 수의 개수만큼 재현이가 외친 수 입력받기
for i in range(K) :
    num = int(sys.stdin.readline())

    if num == 0 :
        stack.pop()
    else :
        stack.append(num)

print(sum(stack))

File: src/baekjoon/step/stackQueueDeque/PROB10773.java
package baekjoon.step.stackQueueDeque;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class PROB10773 {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
    static Stack<Integer> stack = new Stack<>();
    static int K;
    static int number;
    static int index;
    static int sum = 0;

    public static void main(String[] args) throws IOException {
        K = Integer.parseInt(br.readLine());
        for (int i = 0; i < K; i ++) {
            number = Integer.parseInt(br.readLine());
            if (number == 0) {
                stack.pop();
            } else {
                stack.add(number);
            }
        }

        index = stack.size();

        for (int i = 0; i < index; i ++) {
            sum += stack.pop();
        }

        System.out.println(sum);
    }
}
